forge-game/src/main/java/forge/game/spellability/SpellAbilityRestriction.java

```diff
@@ -474,11 +475,11 @@ public final boolean checkOtherRestrictions(final Card c, final SpellAbility sa,
         }
 
         if (sa.isPwAbility()) {
-            final int initialLimit = c.hasKeyword("CARDNAME's loyalty abilities can be activated twice each turn rather than only once") ? 1 : 0;
-            final int limits = c.getAmountOfKeyword("May activate CARDNAME's loyalty abilities once") + initialLimit;
+            final int initialLimit = StaticAbilityNumLoyaltyAct.limitIncrease(c) ? 1 : 0;
+            final int limit = StaticAbilityNumLoyaltyAct.additionalActivations(c) + initialLimit;
```

Comment:
> In some unusual cases, Comet may gain loyalty abilities other than the one he normally has. In those cases, the effect of rolling a 6 applies only to his normal ability.

I think for this you can add a param that checks if `getEffectSource()` root from static host is the sa here